src/serac/numerics/functional/functional.hpp

```diff
@@ -322,8 +322,8 @@ class Functional<test(trials...), exec> {
    * @note The @p Dimension parameters are used to assist in the deduction of the @a geometry_dim
    * and @a spatial_dim template parameter
    */
-  template <int dim, int... args, typename lambda>
-  void AddBoundaryIntegral(Dimension<dim>, DependsOn<args...>, lambda&& integrand, mfem::Mesh& domain)
+  template <int dim, int... args, typename Integrand>
+  void AddBoundaryIntegral(Dimension<dim>, DependsOn<args...>, Integrand&& integrand, mfem::Mesh& domain)
```

Comment:
Same here. You either need to forward integrand on line 335 or use a const lvalue reference instead.

Reply:
There are a bunch of places in this file where this antipattern occurs.